src/containers/ProductSearch/index.jsx

```diff
@@ -25,10 +28,24 @@ class ProductSearch extends Component {
     };
     this.kvass = new Kvass();
     this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
+
+    this.store = UserStore;
+    this.storeKeys = ['apiUser'];
   }
 
+  // Get personalized products on first load, based on general populariy or based on user if logged in
   componentDidMount() {
-    this.searchProduct('');
+    this.setState({isLoading: true});
+    const userId = !this.state.apiUser ? '' : this.state.apiUser._id.$oid;
+    const body = {model_type: 'collaborative_filtering_recommender', source_id: userId, source: 'user', destination: 'product', size: this.pagination.size};
+    this.setState({isLoading: true});
```

Comment:
Can we set the state of loading to true in the constructor, to avoid re-render?

Reply:
fixing